src/backend/distributed/sql/udfs/worker_check_connection_to_node/latest.sql

```diff
@@ -0,0 +1,10 @@
+CREATE FUNCTION pg_catalog.worker_check_connection_to_node(nodename text,
```

Comment:
why do we have both return and OUT? Do you have any specific scenario in mind?  I think RETURN should be enough?

Reply:
Also, maybe use DEFAULT 5432 for the port

Reply:
> why do we have both return and OUT? Do you have any specific scenario in mind? I think RETURN should be enough?
I wanted to have a reasonable tuple descriptor, and that's why I have OUT parameters. I can drop the omit the RETURN clause as it is only there for readability purposes. IMO it is easier to understand the return type if it is spelled out explicitly. The alternative is to go through the arguments and check all OUT or INOUT parameters.

[CREATE FUNCTION docs](https://www.postgresql.org/docs/current/sql-createfunction.html) contain the following paragraph: 
>  When there are OUT or INOUT parameters, the RETURNS clause can be omitted. If present, it must agree with the result type implied by the output parameters: RECORD if there are multiple output parameters, or the same type as the single output parameter.

> Also, maybe use DEFAULT 5432 for the port
Nice idea 👍🏽

Reply:
For such a simple function, I'd probably avoid the `OUT` parameter and simply rely on the return value, but if you feel strong about OUT parameter, yeah why not.